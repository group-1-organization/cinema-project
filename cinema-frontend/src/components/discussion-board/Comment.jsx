import moment from 'moment';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

const Comment = ({ comment }) => {
    return (
        <div className="border-bottom m-3">
            <Row>
                <Col>
                    <p>
                        <strong>{comment.author}</strong>
                    </p>
                </Col>
                <Col>
                    <small className="float-right">{moment(comment.createdAt).fromNow()}</small>
                </Col>
            </Row>
            <Row>
                <Col>
                    <p>
                        {comment.message}
                    </p>
                </Col>
            </Row>
        </div>
    )
}

export default Comment
