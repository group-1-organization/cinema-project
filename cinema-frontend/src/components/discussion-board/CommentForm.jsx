import { useState } from 'react';
import axios from 'axios';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
const Filter = require('bad-words'), filter = new Filter();

const CommentForm = ({ topic, setCommentsLoaded }) => {

    const [author, setAuthor] = useState('');
    const [message, setMessage] = useState('');

    const handleClick = (event) => {
        event.preventDefault();
        const comment = { "topicId": topic._id, "author": filter.clean(author), "message": filter.clean(message) };

        axios.post(`http://localhost:5000/cinema/comments/`, comment).then((response) => {
            setCommentsLoaded(false);
            setAuthor('');
            setMessage('');
        })
    }

    return (
        <Form>
            <Form.Group controlId="authorInput">
                <Form.Label>Your Name</Form.Label>
                <Form.Control type="text" placeholder="Enter your name" value={author} onChange={(event) => setAuthor(event.target.value)} required pattern=".*\S+.*" title="This field is required" />
                <Form.Control.Feedback type="invalid">
                    Please choose a username.
                </Form.Control.Feedback>
            </Form.Group>
            <Form.Group controlId="commentTextarea">
                <Form.Label>Message</Form.Label>
                <Form.Control as="textarea" rows={3} value={message} onChange={(event) => setMessage(event.target.value)} required pattern=".*\S+.*" title="This field is required" />
            </Form.Group>
            <Button variant="primary" type="Submit" onClick={handleClick}>
                Submit
            </Button>
        </Form>
    )
}

export default CommentForm
