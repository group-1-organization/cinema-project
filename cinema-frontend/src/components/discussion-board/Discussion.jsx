import { useState } from 'react';
import Container from 'react-bootstrap/Container';
import Topics from './Topics';
import TopicForm from './TopicForm';

const Discussion = () => {

    const [topicsLoaded, setTopicsLoaded] = useState(false);

    return (
        <div>
            <div className="page-header">
                <h1>Discussion board</h1>
            </div>

            <Container>
                <h3>Topics</h3>
                <TopicForm setTopicsLoaded={setTopicsLoaded} />
                <Topics topicsLoaded={topicsLoaded} setTopicsLoaded={setTopicsLoaded} />
            </Container >
        </div>
    )
}

export default Discussion
