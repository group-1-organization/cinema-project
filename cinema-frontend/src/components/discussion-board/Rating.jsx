import { useState, useEffect } from 'react';
import axios from 'axios';
import StarRatings from 'react-star-ratings';

const Rating = ({ topic, ratingLoaded, setRatingLoaded }) => {

    const [movieRating, setMovieRating] = useState({});

    const getMovieRating = async () => {
        await axios.get(`http://localhost:5000/cinema/ratings/movie/${topic.movieName}`).then((response) => {
            setMovieRating(response.data[0]);
            setRatingLoaded(true);
        })
    }

    useEffect(() => {
        if (topic.movieName !== undefined) {
            getMovieRating();
        }
    }, [ratingLoaded, topic]);

    if (movieRating === undefined) {
        return (
            <p>No Ratings</p>
        )
    } else if (!ratingLoaded) {
        return (
            <p>Loading...</p>
        )
    } else {
        return (
            <div>
                <p>Current average rating by {movieRating.count} people</p>
                <StarRatings
                    rating={movieRating.avgRating}
                    starDimension="40px"
                    starRatedColor="gold"
                />
            </div >
        )
    }

}

export default Rating
