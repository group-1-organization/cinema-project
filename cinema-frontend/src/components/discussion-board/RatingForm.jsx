import { useState } from 'react';
import axios from 'axios';
import Button from 'react-bootstrap/Button';
import StarRatings from 'react-star-ratings';

const RatingForm = ({ topic, setRatingLoaded }) => {

    const [stars, setStars] = useState(0);

    const submitRating = () => {
        if (stars !== 0) {
            let rating = {
                movieName: topic.movieName,
                rating: stars
            };
            axios.post(`http://localhost:5000/cinema/ratings/`, rating).then((response) => {
                setRatingLoaded(false);
                setStars(0);
            })
        }
    }

    return (
        <>
            <StarRatings
                rating={stars}
                starRatedColor="gold"
                starHoverColor="gold"
                changeRating={setStars}
                numberOfStars={5}
                name="rating"
                starDimension="1em"
                starSpacing="0px"
            />
            <Button onClick={submitRating} className="m-3">Submit</Button>
        </>
    )
}

export default RatingForm
