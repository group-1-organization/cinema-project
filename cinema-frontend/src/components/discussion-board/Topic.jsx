import Container from 'react-bootstrap/Container';
import { useParams } from "react-router-dom";

const Topic = () => {

    const { id } = useParams();

    return (
        <Container>
            <h1>Topic</h1>
            <p>{id}</p>
        </Container>
    )
}

export default Topic
