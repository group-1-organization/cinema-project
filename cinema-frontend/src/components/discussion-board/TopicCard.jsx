import Card from 'react-bootstrap/Card';
import moment from 'moment';
import { Link } from "react-router-dom";

const TopicCard = ({ topic }) => {
    return (
        <Card bg="dark" className="mt-3">
            <Card.Header>{topic.movieName}</Card.Header>
            <Card.Body>
                <blockquote className="blockquote mb-0">
                    <Link to={`/topic/${topic._id}`}>
                        <p>{topic.name}</p>
                    </Link>
                    <footer className="blockquote-footer">
                        created {moment(topic.createdAt).fromNow()} by {topic.author}
                    </footer>
                </blockquote>
            </Card.Body>
        </Card>
    )
}

export default TopicCard
