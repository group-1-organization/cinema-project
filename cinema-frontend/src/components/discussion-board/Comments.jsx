import { useState, useEffect } from 'react';
import axios from 'axios';

import Comment from './Comment';

const Comments = ({ topic, commentsLoaded, setCommentsLoaded }) => {

    const [comments, setComments] = useState([]);

    const getComments = async () => {
        await axios.get(`http://localhost:5000/cinema/comments/${topic._id}`).then((response) => {
            setCommentsLoaded(true);
            setComments(response.data);
        })
    }

    useEffect(() => {
        if (topic !== undefined) {
            getComments();
        }
    }, [commentsLoaded]);

    return (
        <>
            {comments.map((comment, index) => (
                <Comment key={index} comment={comment} />
            ))}
        </>
    )
}

export default Comments
