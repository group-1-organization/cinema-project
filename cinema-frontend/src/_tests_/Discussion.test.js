import renderer, { act } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Discussion from '../components/discussion-board/Discussion';
import Topic from '../components/discussion-board/Topic';
import Comment from '../components/discussion-board/Comment';
import CommentForm from '../components/discussion-board/CommentForm';
import Comments from '../components/discussion-board/Comments';
import Rating from '../components/discussion-board/Rating';
import RatingForm from '../components/discussion-board/RatingForm';
import TopicCard from '../components/discussion-board/TopicCard';
import TopicForm from '../components/discussion-board/TopicForm';
import Topics from '../components/discussion-board/Topics';

test('Discussion page matches snapshot', () => {
    const tree = renderer.create(<Discussion />);

    expect(tree).toMatchSnapshot();
});

// test('Topic matches snapshot', () => {
//   const tree = renderer.create(<Topic />);

//   expect(tree).toMatchSnapshot();
// });


test('Comment matches snapshot', () => {

    const comment = {
        _id: "60eee726bef3502694dca9f5",
        topicId: "60eee716bef3502694dca9ea",
        message: "Comments ****",
        author: "Amandeep",
        createdAt: "2021-07-14T13:31:18.612Z",
        updatedAt: "2021-07-14T13:31:18.612Z"

    }
    const tree = renderer.create(<Comment comment={comment} />);

    expect(tree).toMatchSnapshot();
});

test('Comment form matches snapshot', () => {
    const tree = renderer.create(<CommentForm />);

    expect(tree).toMatchSnapshot();
});

test('Comments matches snapshot', () => {
    const tree = renderer.create(<Comments />);

    expect(tree).toMatchSnapshot();
});

test('Rating matches snapshot', () => {
    const tree = renderer.create(<Rating />);

    expect(tree).toMatchSnapshot();
});

// test('Rating form matches snapshot', () => {

//   const topic = {
//     _id: "60eee716bef3502694dca9ea",
//     name: "New components",
//     movieName: "Fast & Furious 9",
//     author: "Amandeep",
//     createdAt: "2021-07-14T13:31:02.653Z",
//     updatedAt: "2021-07-14T13:31:02.653Z"
//   }
//   const tree = renderer.create(<RatingForm topic={topic} />);

//   expect(tree).toMatchSnapshot();
// });

// test('TopicCard matches snapshot', () => {

//   const topic = {
//     _id: "60eee716bef3502694dca9ea",
//     name: "New components",
//     movieName: "Fast & Furious 9",
//     author: "Amandeep",
//     createdAt: "2021-07-14T13:31:02.653Z",
//     updatedAt: "2021-07-14T13:31:02.653Z"

//   }
//   const tree = renderer.create(<Router><TopicCard topic={topic} /></Router>);

//   expect(tree).toMatchSnapshot();
// });

// test('TopicForm matches snapshot', () => {

//     // act(() => {
//     //   const topic = {
//     //     _id: "60eecf74bef3502694dca968",
//     //     name: "2nd Topic",
//     //     movieName: "In The Heights",
//     //     author: "Scott",
//     //     // createdAt: "2021-07-14T11:50:12.108Z",
//     //     // updatedAt: "2021-07-14T11:50:12.108Z"

//     //   }
//     //   const tree = renderer.create(<TopicForm topic={topic} />);
//     // })

//     const topic = {
//         _id: "60eecf74bef3502694dca968",
//         name: "2nd Topic",
//         movieName: "In The Heights",
//         author: "Scott",
//         createdAt: "2021-07-14T11:50:12.108Z",
//         updatedAt: "2021-07-14T11:50:12.108Z"

//     }
//     const tree = renderer.create(<TopicForm topic={topic} />);

//     expect(tree).toMatchSnapshot();
// });

// test('Topics matches snapshot', () => {


//   const tree = renderer.create(<Topics />);

//   expect(tree).toMatchSnapshot();
// });