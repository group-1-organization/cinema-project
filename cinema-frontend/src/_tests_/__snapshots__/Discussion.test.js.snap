// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment form matches snapshot 1`] = `
<form
  className=""
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="authorInput"
    >
      Your Name
    </label>
    <input
      className="form-control"
      id="authorInput"
      onChange={[Function]}
      pattern=".*\\\\S+.*"
      placeholder="Enter your name"
      required={true}
      title="This field is required"
      type="text"
      value=""
    />
    <div
      className="invalid-feedback"
    >
      Please choose a username.
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="commentTextarea"
    >
      Message
    </label>
    <textarea
      className="form-control"
      id="commentTextarea"
      onChange={[Function]}
      pattern=".*\\\\S+.*"
      required={true}
      rows={3}
      title="This field is required"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="Submit"
  >
    Submit
  </button>
</form>
`;

exports[`Comment matches snapshot 1`] = `
<div
  className="border-bottom m-3"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        <strong>
          Amandeep
        </strong>
      </p>
    </div>
    <div
      className="col"
    >
      <small
        className="float-right"
      >
        a few seconds ago
      </small>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        Comments ****
      </p>
    </div>
  </div>
</div>
`;

exports[`Comments matches snapshot 1`] = `null`;

exports[`Discussion page matches snapshot 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1>
      Discussion board
    </h1>
  </div>
  <div
    className="container"
  >
    <h3>
      Topics
    </h3>
    <button
      className="button-colour btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Create topic
    </button>
  </div>
</div>
`;

exports[`Rating matches snapshot 1`] = `
<p>
  Loading...
</p>
`;
